app/qml/map/MMMapController.qml

```diff
@@ -795,39 +798,56 @@ Item {
     }
   }
 
-  MMListDrawer {
-    id: activeLayerPanel
+  Loader {
+    id: activeLayerPanelLoader
+    active: false
+    sourceComponent: activeLayerPanelComponent
+  }
 
-    drawerHeader.title: qsTr( "Choose Active Layer" )
+  Component {
+    id: activeLayerPanelComponent
 
-    list.model: __recordingLayersModel
+    MMListDrawer {
+      id: activeLayerPanel
 
-    list.delegate: MMListDelegate {
-      text: model.layerName
+      drawerHeader.title: qsTr( "Choose Active Layer" )
 
-      // TODO: why we need to set hight here?
-      height: __style.menuDrawerHeight
+      list.model: MM.LayersProxyModel {
+        id: recordingLayersModel
 
-      leftContent: MMIcon {
-        source: model.iconSource
+        qgsProject: __activeProject.qgsProject
+        modelType: MM.LayersProxyModel.ActiveLayerSelection
+        model: MM.LayersModel {}
       }
 
-      rightContent: MMIcon {
-        source: __style.doneCircleIcon
-        visible: __activeLayer.layerId === model.layerId
-      }
+      list.delegate: MMListDelegate {
+        text: model.layerName
 
-      onClicked: {
-        __activeProject.setActiveLayer( __recordingLayersModel.layerFromLayerId( model.layerId ) )
-        activeLayerPanel.close()
+        // TODO: why we need to set hight here?
+        height: __style.menuDrawerHeight
+
+        leftContent: MMIcon {
+          source: model.iconSource
+        }
+
+        rightContent: MMIcon {
+          source: __style.doneCircleIcon
+          visible: __activeLayer.layerId === model.layerId
+        }
+
+        onClicked: {
+          __activeProject.setActiveLayer( recordingLayersModel.layerFromLayerId( model.layerId ) )
+          activeLayerPanel.close()
+          activeLayerPanelLoader.active = false
```

Comment:
We also need to deactivate the loader when closing the drawer without clicking on any delegate